### PR TITLE
Fix nesting of RedisCreateProperties inside RedisResourceProperties

### LINKED ISSUE #732: [Redis] x-ms-client-flattening is used incorrectly
Regarding issue https://github.com/Azure/azure-sdk-for-go/issues/459 found by @tombuildsstuff 

cc:
@TimLovellSmith

### PR DESCRIPTION
**Pull request by TimLovellSmith**
To fix #732.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] Validation errors from the [Linter extension for VS Code](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/linter.md) have all been fixed for this spec. (**Note:** for large, previously checked in specs, there will likely be many errors shown. Please contact our team so we can set a timeframe for fixing these errors if your PR is not going to address them).
  - [x] The spec follows the patterns described in the [Swagger good patterns](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-good-patterns.md) document unless the service API makes this impossible.
…sourceProperties (with "x-ms-client-flatten": true).

---
This change is [Reviewable](https://reviewable.io/reviews/azure/azure-rest-api-specs/746)

### CODE REVIEW

File: arm-redis/2016-04-01/swagger/redis.json [hunk 1]
```diff
@@ -707,12 +711,6 @@
       "x-ms-external": true
     },
     "RedisCreateProperties" : {
-      "properties": {
-        "sku": {
-          "$ref": "#/definitions/Sku",
-          "description": "The SKU of the Redis cache to deploy."
-        }
-      },
       "required": [
```
Comment (fearthecowboy) [inline arm-redis/2016-04-01/swagger/redis.json]: You can't change a property from an `allOf` composition to `required` . 
(The property won't actually get marked as required at all.)

**Review by fearthecowboy — COMMENTED** [review]
Just the one little issue found. I'm not sure there is a good way to accurately represent what you're trying to do with that inherited `sku` property. 

Gimme a little bit here, I'm going to dig into it and see if this is really just an AutoRest limitation, or if swagger does support it.

File: arm-redis/2016-04-01/swagger/redis.json (hunk 1 above)
Comment (TimLovellSmith) [inline arm-redis/2016-04-01/swagger/redis.json]: I guess I'll just have to move it to the derived types then.

Comment (TimLovellSmith) [inline arm-redis/2016-04-01/swagger/redis.json]: I guess I'll just have to move it to the derived types then.

**Comment by fearthecowboy** [comment]
I'm happy enough with this; @TimLovellSmith if you're done, I can commit it.

**Review by fearthecowboy — APPROVED** [review]
LGTM

**Comment by TimLovellSmith** [comment]
@fearthecowboy Yup! I think I'm done for this PR.

**Comment by amarzavery** [comment]
@fearthecowboy - This PR was not ready to be merged. We found lot of issues such as
- PascalCase naming of redisConfiguration and redisVersion
- Properties being modeled incorrectly (properties.properties)
- shardCount type was modeled as string whereas it was integer on wire
I had asked @TimLovellSmith to provide examples in "x-ms-examples" format. This conversation happened over email and I forgot to add you (@fearthecowboy). Sorry about that.

@TimLovellSmith - Can you please provide us examples in "x-ms-examples" format for every operation in the Redis swagger? This would help your team and us in serving our customers in a better way (less breaking changes and hot fixes).

**Comment by TimLovellSmith** [comment]
@amarzavery I fixed casing of redisConfiguration and redisVersion as part of the swagger PR. shardCount is also already modeled as an int in swagger, isn't it?

**Comment by TimLovellSmith** [comment]
@amarzavery I fixed casing of redisConfiguration and redisVersion as part of the swagger PR. shardCount is also modeled as an int correctly in swagger. Oh wait... there's a naming bug. 'shardId'?

**Pull request merged**